### PR TITLE
Migrate do test clients

### PR DESCRIPTION
**Pull request by obabec**
### Type of change

- Refactoring

### Description
Refactoring continue on doTest methods with client uses. Migrated rest of doTest methods from current tests to clientRunner.

### Checklist

- [ ] Update/write design documentation in `./documentation/design`
- [ ] Write tests and make sure they pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

File: systemtests/src/test/java/io/enmasse/systemtest/TestBase.java [hunk 1]
```diff
@@ -86,4 +89,18 @@ void cleanup() throws Exception {
             throw exception;
         }
     }
+
+    protected void assertDefaultMessaging() {
```
Comment (kornys) [inline systemtests/src/test/java/io/enmasse/systemtest/TestBase.java:93]: remove this. Test base is not for storing assertMethods anymore

File: systemtests/src/main/java/io/enmasse/systemtest/messagingclients/MessagingClientRunner.java [hunk 2]
```diff
@@ -105,6 +119,69 @@ public void sendAndReceive(MessagingEndpoint endpoint, boolean waitReceivers, Ma
         }
     }
 
+    public void endpointSendAndReceiveOnCluster(String host, int port, String address, boolean enableTls, boolean websockets) throws InterruptedException {
```
Comment (kornys) [inline systemtests/src/main/java/io/enmasse/systemtest/messagingclients/MessagingClientRunner.java:122]: please change name, I don't understand why it is named endpointSendEndReceive

File: systemtests/src/test/java/io/enmasse/systemtest/TestBase.java (hunk 1 above)
Comment (obabec) [reply systemtests/src/test/java/io/enmasse/systemtest/TestBase.java:93]: Where you suggest to store this method afterwards? I don't see point why common assertMethod which is used in most of the tests shouldn't be there, moving it somewhere like testUtils is not right in my opinion.

Comment (kornys) [reply systemtests/src/test/java/io/enmasse/systemtest/TestBase.java:93]: AssertionUtils is a best approach I guess, create static class and store special Asserts there

Comment (kornys) [reply systemtests/src/test/java/io/enmasse/systemtest/TestBase.java:93]: and TestBase is not for storing common methods, we have testbase only for extend junit extension and configuration and basic clean in some case

File: systemtests/src/main/java/io/enmasse/systemtest/messagingclients/MessagingClientRunner.java (hunk 2 above)
Comment (obabec) [reply systemtests/src/main/java/io/enmasse/systemtest/messagingclients/MessagingClientRunner.java]: Changed, but I am thinking that we could maybe squash sendAndReceive with this method in some way. Need to take a look on that little bit more.

File: systemtests/src/test/java/io/enmasse/systemtest/TestBase.java (hunk 1 above)
Comment (obabec) [reply systemtests/src/test/java/io/enmasse/systemtest/TestBase.java]: OK, thanks for explanation of current situation, I have create AssertionUtils for these as you suggested. Although I think that we should use as least static classes as possible.

**Comment by obabec** [comment]
@lulf Sure they are and I think in process of creating new tests we will come up with options we want/need to configure and we will change functions to handle them :)

**Pull request merged**